framework/Databases.py

```diff
@@ -292,193 +293,186 @@ def __retrieveDataPointSet(self,attributes):
+
         if allOutParam:
           for key in histVar[1]['outputSpaceHeaders']:
             if i == 0:
               outDict[key] = np.zeros(len(histList))
-            outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)
+            outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(key)]
         else:
           for key in outParam:
-            if i == 0:
-              outDict[key] = np.zeros(len(histList))
             if key in histVar[1]['outputSpaceHeaders'] or \
                utils.toBytes(key) in histVar[1]['outputSpaceHeaders']:
+              if i == 0:
+                outDict[key] = np.zeros(len(histList))
               if key in histVar[1]['outputSpaceHeaders']:
-                outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)
+                outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(key)]
               else:
-                outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(utils.toBytes(key))], kind='linear')(outputPivotVal)
+                outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(utils.toBytes(key))]
             else:
-              self.raiseAnError(IOError,"the parameter " + key + " has not been found")
+              self.raiseAnError(IOError,"the parameter " + str(key) + " has not been found")
+
+      # else:
+      #   # Arbitrary point in outputPivotVal case... If the requested outputPivotVal point Set does not match any of the stored ones and
+      #   # start_outputPivotVal <= requested_outputPivotVal_point <= end_outputPivotVal, compute an interpolated value
+      #   if allOutParam:
+      #     for key in histVar[1]['outputSpaceHeaders']:
+      #       if i == 0:
+      #         outDict[key] = np.zeros(len(histList))
+      #       outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)
+      #   else:
+      #     for key in outParam:
+      #       if i == 0:
+      #         outDict[key] = np.zeros(len(histList))
+      #       if key in histVar[1]['outputSpaceHeaders'] or \
+      #          utils.toBytes(key) in histVar[1]['outputSpaceHeaders']:
+      #         if key in histVar[1]['outputSpaceHeaders']:
+      #           outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)
+      #         else:
+      #           outDict[key][i] = interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(utils.toBytes(key))], kind='linear')(outputPivotVal)
+      #       else:
+      #         self.raiseAnError(IOError,"the parameter " + key + " has not been found")
```

Comment:
Do you want to keep the commented lines?

Reply:
removed